src/main/java/Duke.java

```diff
@@ -1,10 +1,77 @@
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
+
+    public static void greet_user() {
+        System.out.println("Hello! I'm Duke \n");
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println(logo);
+        System.out.println("How can i help u? \n");
+    }
+
+    public static void echo() {
+        Scanner scan = new Scanner(System.in);
+
+        String input = scan.next();
+        if ("bye".equals(input)) {
+            System.out.println("Goodbye. Hope to see u again :) \n");
+        } else {
+            System.out.println(input);
+            echo();
+        }
+
+        scan.close();
+
+    }
+
+    public static void main(String[] args) {
+        greet_user();
+        int counter = 0;
+
+        Tasks[] list_of_tasks = new Tasks[101]; // Following the assumption that there will be no more than 100 tasks
+        for (int k = 0; k < 101; k++) {
+            list_of_tasks[k] = new Tasks("");
+        }
+
+        Scanner scan = new Scanner(System.in);
+        String input = scan.nextLine();
+
+        while (!"bye".equals(input)) {
+            if ("list".equals(input)) {
+                System.out.println("Here are the tasks in your list: ");
+                for (int i = 0; i < counter; i++) {
+                    System.out.println(
+                            i + 1 + "." + " " + list_of_tasks[i].getStatusIcon() + " " + list_of_tasks[i].description);
+                }
```

Comment:
It may be clearer to extract these parts as separate functions to avoid nesting loops.